uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/ContactTest.php

```diff
@@ -190,4 +187,78 @@ public function testGetLeaveManageesReturnsEmptyResultsWhenStaffMemberIsTryingTo
+    $this->callAPI('Contact', 'getleavemanagees', ['unassigned' => false]);
+  }
+
+  /**
+   * @expectedExceptionMessage Unassigned cannot be true and managed_by parameter also present
+   * @expectedException CiviCRM_API3_Exception
+   */
+  public function testGetLeaveManageesThrowsAnExceptionWhenUnassignedIsTrueAndManagedByParameterIsPresent() {
+    $this->callAPI('Contact', 'getleavemanagees', ['unassigned' => true, 'managed_by' => 1]);
+  }
+
+  /**
+   * @expectedExceptionMessage Either unassigned must be true or managed_by parameter present
+   * @expectedException CiviCRM_API3_Exception
+   */
+  public function testGetLeaveManageesThrowsAnExceptionWhenUnassignedAndManagedByParameterIsAbsent() {
+    $this->callAPI('Contact', 'getleavemanagees', []);
+  }
+
+  public function testGetLeaveManageesReturnsOnlyContactsWithoutActiveLeaveApproverRelationshipWhenUnassignedTrue() {
+    $contact1 = $this->contact1;
+    $contact2 = $this->contact2;
+    $contact3 = ContactFabricator::fabricate();
+    $manager = $this->manager;
+
+    $relationshipType = RelationshipTypeFabricator::fabricate();
+    RelationshipFabricator::fabricate([
+      'contact_id_a' => $contact3['id'],
+      'contact_id_b' => $manager['id'],
+      'relationship_type_id' => $relationshipType['id']
+    ]);
+
+    $this->setContactAsLeaveApproverOf($manager, $contact1);
+
+    $result = $this->callAPI('Contact', 'getleavemanagees', ['unassigned' => true]);
+
+    //The manager, contact2 and contact3 does not have an active leave approver
+    //relationship.
+    $this->assertEquals(3, $result['count']);
+    $this->assertEquals($result['values'][$contact2['id']]['id'], $contact2['id']);
+    $this->assertEquals($result['values'][$contact3['id']]['id'], $contact3['id']);
+    $this->assertEquals($result['values'][$manager['id']]['id'], $manager['id']);
+  }
+
+  public function testGetLeaveManageesReturnsNoInformationForContactWithActiveLeaveManagerAndOtherRelationshipWhenUnassignedIsTrue() {
+    $contact1 = $this->contact1;
+    $contact2 = $this->contact2;
+    $manager = $this->manager;
+
+    $this->setContactAsLeaveApproverOf($manager, $contact1);
+    $relationshipType = RelationshipTypeFabricator::fabricate();
+
+    //Add a neutral relationship between contact1 and manager that is not of
+    //type leave approver.
+    RelationshipFabricator::fabricate([
+      'contact_id_a' => $contact1['id'],
+      'contact_id_b' => $manager['id'],
+      'relationship_type_id' => $relationshipType['id']
+    ]);
+
+    $result = $this->callAPI('Contact', 'getleavemanagees', ['unassigned' => true]);
+
+    //contact2 and and manager will be returned because they are without active leave
+    //approver relationdip.
+    $this->assertEquals(2, $result['count']);
+    $this->assertEquals($result['values'][$contact2['id']]['id'], $contact2['id']);
+    $this->assertEquals($result['values'][$manager['id']]['id'], $manager['id']);
+  }
 }
```

Comment:
What about adding tests for when you're logged in as admin and using unassigned = true?